crates/turborepo-paths/src/absolute_system_path.rs

```diff
@@ -103,8 +105,8 @@ impl AbsoluteSystemPath {
         }
     }
 
-    pub(crate) fn new_unchecked(path: &Path) -> &Self {
-        unsafe { &*(path as *const Path as *const Self) }
+    pub unsafe fn new_unchecked(path: &Path) -> &Self {
```

Comment:
Can you use `AbsoluteSystemPathBuf::new().unwrap()` instead? It looks like this is only used in tests